cdc/log.cc

```diff
@@ -1504,22 +1504,27 @@ class transformer final : public change_processor {
 
         assert(_builder);
 
-        auto image_ck = _builder->allocate_new_log_row(op);
-        if (ck) {
-            _builder->set_clustering_columns(image_ck, *ck);
-        }
-
         const auto kind = ck ? column_kind::regular_column : column_kind::static_column;
 
         cell_map* row_state;
         if (ck) {
             row_state = get_row_state(_clustering_row_states, *ck);
             if (!row_state) {
                 // We have no data for this row, we can stop here
+                // Empty row here means we had data, but column deletes
+                // removed it. This we should report as pk/ck only
                 return;
             }
         } else {
             row_state = &_static_row_state;
+            if (row_state->empty()) {
```

Comment:
I understand the difference between row_state == nullptr and row_state->empty() == true for clustering rows. First means there was no row or the row was deleted. Second means the row has only empty cells. Right?

What's the story with static row? Doesn't row_state->empty() == true mean removal of it's all columns but not removal of the static row itself?

Reply:
Static row is only really "removed" on PK delete. But we don't really make any distinction on whether static row actually exists (on this level), or if it was emptied. 
The above is probably not really correct. We should maybe add a state var to track if we created/processed static row at all.

Reply:
changed this to check iff we touched static row during processing, and generate empty postimage if so, and none otherwise (if the row state is empty). This should be consistent with "normal" clustered row state.